examples/basic-infection/people.rs

```diff
@@ -0,0 +1,115 @@
+use ixa::context::Context;
+use ixa::define_data_plugin;
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Copy, Serialize, Deserialize)]
+pub enum InfectionStatus {
+    S,
+    I,
+    R,
+}
+
+#[derive(Copy, Clone)]
+pub struct InfectionStatusEvent {
+    pub _prev_status: InfectionStatus,
+    pub updated_status: InfectionStatus,
+    pub person_id: usize,
+}
+
+pub trait ContextPeopleExt {
+    fn create_person(&mut self);
+    fn get_person_status(&mut self, person_id: usize) -> InfectionStatus;
+    fn set_person_status(&mut self, person_id: usize, infection_status: InfectionStatus);
+    fn get_population(&mut self) -> usize;
+}
+
+struct PeopleData {
+    people_map: HashMap<usize, InfectionStatus>,
+}
+// Register the data container in context
+define_data_plugin!(
+    PeoplePlugin,
+    PeopleData,
+    PeopleData {
+        people_map: HashMap::<usize, InfectionStatus>::new()
+    }
+);
+
+impl ContextPeopleExt for Context {
+    fn create_person(&mut self) {
+        let people_data_container = self.get_data_container_mut(PeoplePlugin);
+        let person_id = people_data_container.people_map.len();
+        people_data_container
+            .people_map
+            .insert(person_id, InfectionStatus::S);
+    }
+
+    fn get_person_status(&mut self, person_id: usize) -> InfectionStatus {
```

Comment:
This shouldn't need a mutable reference to context I think, so maybe use`&self` / `get_data_container`